specs/governance/alligator.md

```diff
@@ -0,0 +1,390 @@
+# Alligator
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Interface](#interface)
+  - [Core Functions](#core-functions)
+    - [`subdelegate`](#subdelegate)
+    - [`subdelegateFromToken`](#subdelegatefromtoken)
+    - [`subdelegateBatched`](#subdelegatebatched)
+    - [`afterTokenTransfer`](#aftertokentransfer)
+    - [`migrateAccount`](#migrateaccount)
+  - [Getters](#getters)
+    - [`checkpoints`](#checkpoints)
+    - [`numCheckpoints`](#numcheckpoints)
+    - [`getVotes`](#getvotes)
+    - [`getPastVotes`](#getpastvotes)
+    - [`getPastTotalSupply`](#getpasttotalsupply)
+    - [`subdelegations`](#subdelegations)
+  - [Events](#events)
+    - [`Subdelegation`](#subdelegation)
+    - [`Subdelegations`](#subdelegations)
+    - [`DelegateVotesChanged`](#delegatevoteschanged)
+- [Storage](#storage)
+- [Types](#types)
+  - [`SubdelegationRule`](#subdelegationrule)
+  - [`AllowanceType`](#allowancetype)
+- [Backwards Compatibility](#backwards-compatibility)
+- [User Flow](#user-flow)
+  - [Partial delegations](#partial-delegations)
+  - [Constrained delegations](#constrained-delegations)
+  - [Redelegations](#redelegations)
+  - [Differences](#differences)
+- [Security Considerations](#security-considerations)
+  - [Dependence on Alligator](#dependence-on-alligator)
+  - [Connection with GovernanceToken](#connection-with-governancetoken)
+- [Future Considerations](#future-considerations)
+  - [Cross Chain Delegations](#cross-chain-delegations)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Constant | Value                                        |
+|----------|----------------------------------------------|
+| Address  | `0x4200000000000000000000000000000000000043` |
+
+The `Alligator` contract implements subdelegations for the [`GovernanceToken`](gov-token.md). Subdelegations enable
+advanced delegation use cases, such as partial, time-constrained & block-based delegations, and relative & fixed allowances.
+
+The `Alligator` contract migrates the delegation state from the `GovernanceToken` to itself
+through a hook-based approach. Specifically, the `GovernanceToken` calls the `Alligator` contract's
+`afterTokenTransfer` function after a token transfer. This enables the `Alligator` contract to consume the hook and update
+its delegation and checkpoint mappings accordingly. If either address involved in the transfer (`_from_` or `_to`) has not
+been migrated to the `Alligator` contract, the contract copies the address' checkpoint data from the
+`GovernanceToken` to its own state.
+
+## Interface
+
+### Core Functions
+
+#### `subdelegate`
```

Comment:
Does this require the user to first delegate to alligator?

Reply:
users don't need to delegate to alligator